homeassistant/components/duosida_ev/sensor.py

```diff
@@ -0,0 +1,216 @@
+        value_fn=lambda data: data.get("current_l2"),
+        entity_registry_enabled_default=False,
+    ),
+    DuosidaSensorEntityDescription(
+        key="current_l3",
+        translation_key="current_l3",
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.get("current_l3"),
+        entity_registry_enabled_default=False,
+    ),
+    DuosidaSensorEntityDescription(
+        key="power",
+        translation_key="power",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.get("power"),
+    ),
+    DuosidaSensorEntityDescription(
+        key="session_energy",
+        translation_key="session_energy",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda data: data.get("session_energy"),
+    ),
+    DuosidaSensorEntityDescription(
+        key="session_time",
+        translation_key="session_time",
+        icon="mdi:timer",
+        native_unit_of_measurement=UnitOfTime.HOURS,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: (
+            round(data.get("session_time", 0) / 60.0, 2)
+            if data.get("session_time")
+            else 0
+        ),
+    ),
+    DuosidaSensorEntityDescription(
+        key="total_energy",
+        translation_key="total_energy",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        icon="mdi:lightning-bolt",
+        value_fn=lambda data: data.get("total_energy"),
+    ),
+    DuosidaSensorEntityDescription(
+        key="temperature",
+        translation_key="temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.get("temperature_station"),
+    ),
+    DuosidaSensorEntityDescription(
+        key="model",
+        translation_key="model",
+        icon="mdi:information",
+        value_fn=lambda data: data.get("model"),
+        entity_registry_enabled_default=False,
+    ),
+    DuosidaSensorEntityDescription(
+        key="manufacturer",
+        translation_key="manufacturer",
+        icon="mdi:factory",
+        value_fn=lambda data: data.get("manufacturer"),
+        entity_registry_enabled_default=False,
+    ),
+    DuosidaSensorEntityDescription(
+        key="firmware",
+        translation_key="firmware",
+        icon="mdi:chip",
+        value_fn=lambda data: data.get("firmware"),
+        entity_registry_enabled_default=False,
+    ),
```

Comment:
We don't want this in the state machine, instead we should set it to the device registry